src/software/ai/navigator/obstacle/obstacle_factory.cpp

```diff
@@ -163,10 +163,12 @@ ObstaclePtr ObstacleFactory::createObstacleFromRectangle(const Rectangle &rectan
 
 ObstaclePtr ObstacleFactory::createObstacle(const Circle &circle)
 {
-    return ObstaclePtr(std::make_shared<CircleObstacle>(CircleObstacle(circle)));
+    return ObstaclePtr(
```

Comment:
I think this could be simplified to `return std::make_shared<GeomObstacle<Circle>>(circle);`. Same for the line below

Reply:
ok fixed it